app/code/community/Bolt/Boltpay/Helper/Data.php

```diff
@@ -190,16 +191,33 @@ public function buildOnCloseCallback($closeCustom, $checkoutType)
                     }
                     ";
                 break;
+            case Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_PRODUCT_PAGE:
+                $quoteId = Mage::getSingleton('checkout/session')->getQuoteId();
+                $successUrl = $this->getMagentoUrl(Mage::getStoreConfig('payment/boltpay/successpage'), array('checkoutType' => $checkoutType, 'session_quote_id' => $quoteId));
+                break;
             case Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_FIRECHECKOUT:
                 $javascript .=
                     "
+                    $closeCustom
                     isFireCheckoutFormValid = false;
                     initBoltButtons();
                     ";
-                break;
-            case Bolt_Boltpay_Block_Checkout_Boltpay::CHECKOUT_TYPE_PRODUCT_PAGE:
-                $quoteId = Mage::getSingleton('checkout/session')->getQuoteId();
-                $successUrl = $this->getMagentoUrl(Mage::getStoreConfig('payment/boltpay/successpage'), array('checkoutType' => $checkoutType, 'session_quote_id' => $quoteId));
+                $closeCustom = '';
+            default:
```

Comment:
can you add comment when execution comes default?

Reply:
yes, I can.